File: netutils/lib/signalhandler.c
#include "signalhandler.h"

static int signalRecieved = 1;

void setupSignal ()
{
	if (signal(SIGINT, signalHandler) == SIG_ERR)
		fatal ("Setting signal handler", strerror(errno));
}

void signalHandler (int signo)
{
	if (signo == SIGINT) {
		signalRecieved = 0;
		info ("ctrl+c pressed");
	}

}

int checkSignal()
{
	return signalRecieved;
}

File: nettools/lib/signalHandler.py
import signal
import screen

# make it a class 

def signalHandler (signal, frame):
	signalFlag = False
	screen.info ("Ctrl+C pressed")

def initSignal():
	global signalFlag 
	signalFlag = True
	signal.signal (signal.SIGINT, signalHandler)

def checkSignal ():
	return signalFlag
